compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -8123,15 +8097,19 @@ private void createLHSSafeNavigation(List<BLangStatement> stmts, BLangExpression
                 literal.value = ERROR_REASON_NULL_REFERENCE_ERROR;
                 literal.type = symTable.stringType;
 
-                BLangInvocation errorCtorInvocation = (BLangInvocation) TreeBuilder.createInvocationNode();
-                errorCtorInvocation.pos = expr.pos;
-                errorCtorInvocation.argExprs.add(literal);
-                errorCtorInvocation.requiredArgs.add(literal);
-                errorCtorInvocation.type = symTable.errorType;
-                errorCtorInvocation.symbol = symTable.errorConstructor;
+                BLangErrorConstructorExpr errorConstructorExpr =
+                        (BLangErrorConstructorExpr) TreeBuilder.createErrorConstructorExpressionNode();
+                BSymbol symbol = symResolver.lookupMainSpaceSymbolInPackage(errorConstructorExpr.pos, env,
+                        names.fromString(""), names.fromString("error"));
+                errorConstructorExpr.type = symbol.type;
+                errorConstructorExpr.pos = expr.pos;
+                List<BLangExpression> positionalArgs = new ArrayList<>();
+                positionalArgs.add(createStringLiteral(errorConstructorExpr.pos,
+                        "{" + RuntimeConstants.MAP_LANG_LIB + "}InvalidKey"));
```

Comment:
Is the arg correct? Shouldn't we use `literal` here?

Reply:
yes. fixed.